### PR TITLE
Email invite link for the Hidden Group Go To the 404 Page

### LINKED ISSUE #2209: Email invite link for the Hidden Group Go To the 404 Page
**Describe the bug**
When inviting someone from the hidden group then when that user opens the email and clicks on the group link it's going to the 404 pages.

**To Reproduce**
https://www.loom.com/share/0a2a57dcc44f42db8d610c2fdcd26311

**Expected behavior**
After sending the invitations the user should able to access the group from the email link.

**Support ticket links**
https://secure.helpscout.net/conversation/1442571194/128671?folderId=3955985

### PR DESCRIPTION
**Pull request by manishpatolia**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2209 .

### How to test the changes in this Pull Request:

1.Create a hidden group
2.Send invitation
3.Check-in mail "visit button"

### Proof Screenshots or Video
The "Visit Group" button will hide when the group type is hidden.

Hidden Group
https://prnt.sc/10s6x1y

Privet/public group
https://prnt.sc/10s6wt7

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-core/classes/class-bp-email-tokens.php [hunk 1]
```diff
@@ -241,15 +241,21 @@ public function token__group_card_small( $bp_email, $formatted_tokens, $tokens )
...
-																		<tbody>
-																			<tr>
-																				<td height="34px" align="right" style="vertical-align: middle;" class="mobile-padding-bottom">
-																					<a class="mobile-button-center" href="<?php echo bp_get_group_permalink( $group ); ?>" target="_blank" rel="nofollow" style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size: <?php echo esc_attr( floor( $settings['body_text_size'] * 0.875 ) . 'px' ); ?>;text-decoration: none;display: block;border-radius: 100px;text-align: center; height: 16px;line-height: 16px;background: <?php echo $settings['highlight_color']; ?>;color: #fff !important;width: 110px;padding: 8px;"><font style="color:#fff;"><?php _e( 'Visit Group', 'buddyboss' ); ?></font></a>
-																				</td>
-																			</tr>
-																		</tbody>
-																	</table>
+																	<?php 
+																	if ( 'hidden' !== $group->status ){
```
Comment (chetansatasiya) [inline src/bp-core/classes/class-bp-email-tokens.php:245]:
```suggestion
																	if ( 'hidden' !== $group->status ) {
```

File: src/bp-core/classes/class-bp-email-tokens.php [hunk 2]
```diff
@@ -241,15 +241,21 @@ public function token__group_card_small( $bp_email, $formatted_tokens, $tokens )
...
+																	?>
+																		<table cellpadding="0" cellspacing="0" border="0" width="47%" style="width: 47%;" align="right" class="responsive-table">
+																			<tbody>
+																				<tr>
+																					<td height="34px" align="right" style="vertical-align: middle;" class="mobile-padding-bottom">
+																						<a class="mobile-button-center" href="<?php echo bp_get_group_permalink( $group ); ?>" target="_blank" rel="nofollow" style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size: <?php echo esc_attr( floor( $settings['body_text_size'] * 0.875 ) . 'px' ); ?>;text-decoration: none;display: block;border-radius: 100px;text-align: center; height: 16px;line-height: 16px;background: <?php echo $settings['highlight_color']; ?>;color: #fff !important;width: 110px;padding: 8px;"><font style="color:#fff;"><?php _e( 'Visit Group', 'buddyboss' ); ?></font></a>
```
Comment (chetansatasiya) [inline src/bp-core/classes/class-bp-email-tokens.php:251]:
```suggestion
																						<a class="mobile-button-center" href="<?php echo esc_url( bp_get_group_permalink( $group ) ); ?>" target="_blank" rel="nofollow" style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size: <?php echo esc_attr( floor( $settings['body_text_size'] * 0.875 ) . 'px' ); ?>;text-decoration: none;display: block;border-radius: 100px;text-align: center; height: 16px;line-height: 16px;background: <?php echo $settings['highlight_color']; ?>;color: #fff !important;width: 110px;padding: 8px;"><font style="color:#fff;"><?php _e( 'Visit Group', 'buddyboss' ); ?></font></a>
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Need changes

File: src/bp-core/classes/class-bp-email-tokens.php (hunk 2 above)
Comment (manishpatolia) [reply src/bp-core/classes/class-bp-email-tokens.php]: DOne

**Review by chetansatasiya — DISMISSED** [review]

**Comment by sumitbuddyboss** [comment]
Hello @manishpatolia 

I've checked in the fix branch and still, I'm getting the same issue, check the below video.

Video:
https://www.loom.com/share/5605f67e5d5b4aeaa1366366f1990117

Thank you

**Comment by jitendrabanjara1991** [comment]
@chetansatasiya 
Ready for review. Please check it
https://screencast-o-matic.com/watch/crljjYV2byd
cc - @KartikSuthar @sumitbuddyboss @firozwp

**Comment by jitendrabanjara1991** [comment]
@sourabhmatolia 
Here is one question - We don't see hidden group in group directory till member doesn't accept invitation. If member has not accepted group invitation then hidden group's link will redirect to 404. So, As per discussed with @chetansatasiya or @KartikSuthar we will send invitation link in the mail. ( [https://prnt.sc/22b3c5f](https://prnt.sc/22b3c5f) ) - Like this one - http://bbpltform.local/members/bb-arianna/groups/invites/

For more info please check below screenshot.
image

cc - @KartikSuthar @chetansatasiya @firozwp

**Comment by sourabhmatolia** [comment]
@jitendrabanjara1991 

I agree.
For hidden groups we can redirect to a link like this:
`domain.com/members/username/groups/invites/`

Thanks

**Comment by jitendrabanjara1991** [comment]
@KartikSuthar @chetansatasiya 
As per discussed with @sourabhmatolia Button text is fine for hidden .
cc - @firozwp 
Thanks.

**Review by KartikSuthar — APPROVED** [review]

**Comment by jitendrabanjara1991** [comment]
Code status
Merged in dev and qa-ongoing
Thanks

**Comment by sumitbuddyboss** [comment]
Hello @jitendrabanjara1991 @manishpatolia 

**Fixed >**
A related issue is resolved, 

**Testcase - 1**
- Hidden group: Sending the invitations the user should be able to access the group from the email link. - Pass

**Testcase - 2**
-  Private group: Sending the invitations the user should be able to access the group from the email link. - Pass

**Testcase - 3**
- Public group: Sending the invitations the user should be able to access the group from the email link. - Pass

**Testcase - 4**
- Button text is for the hidden group. - Pass


**Video Proof:**
https://www.loom.com/share/9abba8c52b7a4c10aef043e77573c5d1

**Testcase:**
https://docs.google.com/spreadsheets/d/1v7bvYc-n-VZYQu8CcAcc9jLmHFduAQmxz_0n8lhjw7o/edit?pli=1#gid=1451538933

Ready for release 🚀 

Thank you

**Comment by jitendrabanjara1991** [comment]
Code status
Merged in qa-approved
Thanks

**Comment by basil-buddyboss** [comment]
Hello @manishpatolia 

This has been tested in the QA-approved branch including the four test cases and there were no issues encountered.

Result: Pass

**Pull request merged**